WireDomain/Sources/WireDomain/Event Processing/TeamEventProcessor/TeamDeleteEventProcessor.swift

```diff
@@ -30,9 +32,11 @@ protocol TeamDeleteEventProcessorProtocol {
 
 struct TeamDeleteEventProcessor: TeamDeleteEventProcessorProtocol {
 
+    let context: NSManagedObjectContext
+
     func processEvent() async throws {
-        // TODO: [WPB-10184]
-        assertionFailure("not implemented yet")
+        let notification = AccountDeletedNotification(context: context)
```

Comment:
maybe we add a TODO to revisit the implementation: Although the legacy implementation does this, I wonder if our new code should continue to do this.

We basically forward the event to SessionManager to delete the account:

```
  func delete(account: Account, reason: ZMAccountDeletedReason) {
        WireLogger.sessionManager.debug("Deleting account \(account.userIdentifier)...")
        if let secondAccount = accountManager.accounts.first(where: { $0.userIdentifier != account.userIdentifier }) {
            // Deleted an account but we can switch to another account
            select(secondAccount, tearDownCompletion: { [weak self] in
                self?.tearDownSessionAndDelete(account: account)
            })
        } else if accountManager.selectedAccount != account {
            // Deleted an inactive account, there's no need notify the UI
            self.tearDownSessionAndDelete(account: account)
        } else {
            // Deleted the last account so we need to return to the logged out area
            logoutCurrentSession(deleteCookie: true, deleteAccount: true, error: NSError(userSessionErrorCode: .accountDeleted, userInfo: [ZMAccountDeletedReasonKey: reason]))
        }
    }
    ```
I would have not expect this processor to do just the forwarding

Reply:
No, this is not ideal and should be revisited to avoid sending a notification, this is a just a thought but we could consider injecting a SessionManager to the SyncManager (though I don't like the idea of the SyncManager having too much dependencies either..)  and use a callback mechanism to trigger these operations.

Reply:
I will add a TODO for that

Reply:
done, resolving and merging